edith-step/src/components/ReceiptHandler.js

```diff
@@ -0,0 +1,307 @@
+import React from 'react';
+import axios from 'axios';
+
+/**
+ * Displays receipt data from the document ai scan so that the
+ * user can edit it and submit a more accurate copy of it.
+ */
+export default class ReceiptHandler extends React.Component {
+  /**
+   * Constructor
+   * @param {Props} props Setup state.
+   */
+  constructor(props) {
+    super(props);
+    this.state = {userId: '', storeName: '', date: '',
+      name: '', fileUrl: '', totalPrice: 0.0, items: [], deals: []};
+
+    /**
+     * Calculates the current date in yyyy-mm-dd format
+     * @return {string} The current date
+     */
+    this.getDate = () => {
+      const date = new Date(Date.now());
+      let month = date.getMonth() + 1;
+      month < 10 ? month = '0' + month.toString() : month = month.toString();
+      let day = date.getDate();
+      day < 10 ? day = '0' + day.toString() : day = day.toString();
+      const year = date.getFullYear();
+      return [year, month, day].join('-');
+    };
+
+    /**
+     * Handles changes to the grocery store of the trip.
+     * @param {Event} e change event
+     */
+    this.handleStoreChange = (e) => {
+      this.setState({storeName: e.target.value});
+    };
+
+    /**
+     * Adds a form row for the user to input additional grocery items
+     * that weren't included in the initial scan.
+     * @param {Event} e change event
+     */
+    this.addItem = (e) => {
+      const newItem = {
+        userId: this.state.userId,
+        name: '',
+        price: 0.0,
+        quantity: 0,
+        category: '',
+        expiration: '',
+      };
+      this.setState({items: this.state.items.concat(newItem)});
+    };
+
+    /**
+     * Handles primary submission to the form. Sends the grocery
+     * data to the receipt-data servlet to find deals and expiration
+     * dates.
+     *
+     * @param {Event} e change event
+     */
+    this.handleSubmit = async (e) => {
+      e.preventDefault();
+      let price = 0;
+      this.state.items.forEach((item) => {
+        price += item.price * item.quantity;
+      });
```

Comment:
Use `reduce` for this.

```javascript
const price = this.state.items.reduce((sum, item) => sum + item.price * item.quantity, 0);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce